File: HashmapsandSets/JewelandStones.py
class Solution():
    def numJewelsInStones(self, jewels: str, stones: str)-> int:
        """
        :type jewels: str
        :type stones: str
        :rtype: int

        time complexity: O(n + m); where n is amount of stones
        space complexity: O(n); where n is amount of jewels
        """
        jewelSet = set(jewels)
        count = 0
        
        for stone in stones:
            if stone in jewelSet:
                count += 1
        
        return count

File: HashmapsandSets/JewelandStones.java
package HashmapsandSets;
import java.util.HashSet;

public class JewelandStones {

    public static int numJewelsInStones(String jewels, String stones){

        // time complexity: O(n + m); where n is amount of stones and m is the amount of jewels
        // space complexity: O(m): where m is the jewels 
        int count = 0;

        HashSet<Character> jewelSet = new HashSet<>();
        for (char j : jewels.toCharArray()){
            jewelSet.add(j);
        }

        for (char stone : stones.toCharArray()){
            if (jewelSet.contains(stone)){
                count++;
            }
        }
        return count;
    }


    public static void main(String[] args){
        String Jewels = "aA";
        String Stones = "aAAbbbb";

        System.out.println(numJewelsInStones(Jewels, Stones));

    }
    
}
